core/src/main/java/com/sequenceiq/cloudbreak/service/cost/ClusterCostService.java

```diff
@@ -69,23 +79,66 @@ public Map<String, RealTimeCost> getCosts(List<String> clusterCrns, List<String>
                 continue;
             }
 
-            ClusterCostDto clusterCost = instanceTypeCollectorService.getAllInstanceTypes(stack);
+            ClusterCostDto clusterCost = instanceTypeCollectorService.getAllInstanceTypesForCost(stack);
             RealTimeCost realTimeCost = new RealTimeCost();
             realTimeCost.setEnvCrn(stack.getEnvironmentCrn());
             realTimeCost.setResourceCrn(stack.getResourceCrn());
             realTimeCost.setType(stack.getType().name());
             realTimeCost.setResourceName(stack.getName());
+            LOGGER.info("Calculating USD cost for resource {}", stack.getResourceCrn());
             realTimeCost.setHourlyProviderUsd(usdCalculatorService.calculateProviderCost(clusterCost));
             realTimeCost.setHourlyClouderaUsd(usdCalculatorService.calculateClouderaCost(clusterCost, realTimeCost.getType()));
             realTimeCosts.put(stack.getResourceCrn(), realTimeCost);
         }
         return realTimeCosts;
     }
 
+    public Map<String, RealTimeCO2> getCO2(List<String> clusterCrns, List<String> environmentCrns) {
+        errorIfCO2CalculationFeatureIsNotEnabled();
+        Map<String, RealTimeCO2> realTimeCO2Map = new HashMap<>();
+        List<StackView> stacksByCrns = Lists.newArrayList();
+        List<StackView> stackByEnvs = Lists.newArrayList();
+
+        if (CollectionUtils.isNotEmpty(clusterCrns)) {
+            stacksByCrns = stackDtoService.findNotTerminatedByResourceCrnsAndCloudPlatforms(clusterCrns, co2EmissionFactorServiceMap.keySet());
+        }
+        if (CollectionUtils.isNotEmpty(environmentCrns)) {
+            stackByEnvs = stackDtoService.findNotTerminatedByEnvironmentCrnsAndCloudPlatforms(environmentCrns, co2EmissionFactorServiceMap.keySet());
+        }
+
+        List<StackView> stacks = ListUtils.union(stacksByCrns, stackByEnvs);
+        for (StackView stack : stacks) {
+            Status stackStatus = stack.getStatus();
+            if (stackStatus == Status.DELETE_COMPLETED || stackStatus == Status.DELETED_ON_PROVIDER_SIDE) {
```

Comment:
You are fetching findNotTerminatedByResourceCrnsAndCloudPlatforms, NotTerminated clusters, DELETE_COMPLETED is not required here.